packages/application-shell/src/components/navbar/navbar.tsx

```diff
@@ -0,0 +1,678 @@
+        }
+      )}
+    >
+      {props.children}
+    </ul>
+  );
+};
+MenuGroup.displayName = 'MenuGroup';
+
+type MenuItemProps = {
+  hasSubmenu: boolean;
+  isActive: boolean;
+  isMenuOpen: boolean;
+  isBottomItem?: boolean;
+  onClick: MouseEventHandler<HTMLElement>;
+  onMouseEnter?: MouseEventHandler<HTMLElement>;
+  onMouseLeave?: MouseEventHandler<HTMLElement>;
+  children: React.ReactNode;
+};
+const MenuItem = (props: MenuItemProps) => (
+  <li
+    role="menu-item"
+    className={classnames(styles['list-item'], {
+      [styles.item__active]: props.isActive,
+      [styles['item_menu-collapsed']]: !props.isMenuOpen,
+      [styles['item__no-submenu']]: !props.hasSubmenu,
+    })}
+    onClick={props.onClick}
+    onMouseEnter={props.onMouseEnter}
+    onMouseLeave={props.onMouseLeave}
+  >
+    <div className={styles['item-link']}>{props.children}</div>
+  </li>
+);
+MenuItem.displayName = 'MenuItem';
+
+type MenuItemLinkProps = {
+  linkTo?: string;
+  exactMatch: boolean;
+  children: React.ReactNode;
+  onClick?: <TrackFn>(
+    event: SyntheticEvent<HTMLAnchorElement>,
+    track: TrackFn
+  ) => void;
+  useFullRedirectsForLinks?: boolean;
+};
+const menuItemLinkDefaultProps: Pick<MenuItemLinkProps, 'exactMatch'> = {
+  exactMatch: false,
+};
+const MenuItemLink = (props: MenuItemLinkProps) => {
+  const gtmTracking = React.useContext(GtmContext);
+  const redirectTo = (targetUrl: string) => window.location.replace(targetUrl);
+  if (props.linkTo) {
+    return (
+      <NavLink
+        to={props.linkTo}
+        exact={props.exactMatch}
+        activeClassName={styles.highlighted}
+        className={styles['text-link']}
+        onClick={event => {
+          if (props.linkTo && props.useFullRedirectsForLinks) {
+            event.preventDefault();
+            redirectTo(props.linkTo);
+          } else if (props.onClick) {
+            event.persist();
+            props.onClick<typeof gtmTracking.track>(event, gtmTracking.track);
+          }
+        }}
+      >
+        {props.children}
+      </NavLink>
+    );
+  }
+  return <>{props.children}</>;
+};
+MenuItemLink.displayName = 'MenuItemLink';
+MenuItemLink.defaultProps = menuItemLinkDefaultProps;
+
+// This component just render two divs with borders in order to differentiate
+// the settings plugin from the other ones
+const MenuItemDivider = () => (
+  <div className={styles['divider-first-item']}>
+    <div className={styles['divider-second-item']} />
+  </div>
+);
+MenuItemDivider.displayName = 'MenuItemDivider';
+
+// This component wraps `<RestrictedByPermissions>`
+// and the `<ToggleFeature>` components. However, it's necessary to have it as
+// the `<ToggleFeature>` wrapper should be rendered only if the `featureToggle`
+// prop is defined. This is because `<ToggleFeature>` will not render any
+// children if the flag is missing/not found.
+const hasMenuEveryMenuVisibilitySetToBeHidden = (
```

Comment:
maybe should be like `isEveryMenuVisibilitySetToHidden', i find it hard to understand.

Reply:
Good point. I think that initial name is my bad :)